### PR TITLE
Datagrid/nullable enum

### PR DESCRIPTION
**Pull request by DamDam64**
Code Reveue done by objorke

### CODE REVIEW

File: Source/PropertyTools.Wpf/DataGrid/Definitions/PropertyDefinition.cs [hunk 1]
```diff
@@ -299,7 +301,26 @@ public void UpdateDescriptor(Type itemType)
...
         {
-            this.ItemsSource = Enum.GetValues(this.PropertyType);
+            var values = new List<object>();
+            var enumType = this.PropertyType;
+            if (enumType != null)
+            {
+                var ult = Nullable.GetUnderlyingType(enumType);
+                if (ult != null)
+                {
+                    // TODO: this does not work properly, do we need to use a converter?
```
Comment (objorke) [inline Source/PropertyTools.Wpf/DataGrid/Definitions/PropertyDefinition.cs]: do you have a solution for this?

Comment (DamDam64) [inline Source/PropertyTools.Wpf/DataGrid/Definitions/PropertyDefinition.cs]: Have you tried to use my fix, I have commited as the first approach to solve the problem?

From what I can see, no further change is to do to fix the bug.
Please try again and tell me if we have turther problems.
See below.

Regards 
Dietmar

https://github.com/objorke/PropertyTools/pull/101/commits/92f51fa6606c541cab9e39ea1a5b3ebd27c1f6c7

commit 92f51fa6606c541cab9e39ea1a5b3ebd27c1f6c7

         protected void SetEnumItemsSource()
         {
-             this.ItemsSource = Enum.GetValues(this.PropertyType);
+            Type propType = this.PropertyType;
+
+            if (propType.IsGenericType &&
+                propType.GetGenericTypeDefinition() == typeof(Nullable<>))
+            {
+                Type[] typeCol = propType.GetGenericArguments();
+                Type nullableType;
+                if (typeCol.Length > 0)
+                {
+                    nullableType = typeCol[0];
+                    if (nullableType.BaseType == typeof(Enum))
+                    {
+                        this.ItemsSource = Enum.GetValues(nullableType);
+                    }
+                }
+            }
+            else
+            {
+                this.ItemsSource = Enum.GetValues(propType);
+            }
         }

Comment (objorke) [inline Source/PropertyTools.Wpf/DataGrid/Definitions/PropertyDefinition.cs]: I think using `GetUnderlyingType` is cleaner.
The problem is how to show the `null` value in the combo box. Try the example to see the issue. I think a converter or custom item template is needed!

Comment (DamDam64) [inline Source/PropertyTools.Wpf/DataGrid/Definitions/PropertyDefinition.cs]: Hi.

My point of view:
1.) The system should behaive the same way regardless with nullable type to deal with.
Means: if we have a int?, double?, enum?, .. the system must do the same things for rendering, editing, ..
Solution to reach this goal:
https://github.com/objorke/PropertyTools/commit/92f51fa6606c541cab9e39ea1a5b3ebd27c1f6c7

2.) To improve the system is all the time a good idea e.g using custom-converter or custom-templates or what else,
but this is beyong the basic system and located in the customizing corner.

regards Dietmar

**Comment by objorke** [comment]
Hi Dietmar, I have not forgot this PR, will review as soon as I get some time!

**Comment by jogibear9988** [comment]
Maybe use a Nullable Combobox like this for the Enums: https://github.com/icsharpcode/WpfDesigner/blob/master/WpfDesign.Designer/Project/Controls/NullableComboBox.cs

**Comment by objorke** [comment]
Thanks for the link! It looks very interesting!

**Pull request closed without merging**